File: 189 Rotate Array/Approach_01_Brute_force.py
# Approach 01: Brute Force
# Time Complexity: O(N * K)
# Space Complexity: O(1)
def rotate_brute_force(nums: list[int], k: int):
    n = len(nums)
    k %= n  # k-এর মান ঠিক করা
    
    for _ in range(k):
        previous = nums.pop() # শেষ উপাদানটি O(1) সময়ে বের করা
        nums.insert(0, previous) # শুরুতে যোগ করা O(N) সময় নেয়

File: 169 Majority Element/Approach_01_Brute_force.cpp
// Approach 1: Brute Force
// Time Complexity: O(N^2)
// Space Complexity: O(1)

#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int majorityElement(vector<int> &nums)
    {
        int n = nums.size();
        for (int i = 0; i < n; i++)
        {
            int count = 0;
            for(int j = 0; j < n; j++)
            {
                if (nums[j] == nums[i])
                    count++;
            }
            if (count > n / 2)
                return nums[i];
        }
        return -1; // This line will never be reached as per problem constraints    
    }
};